lib/trento_web/views/sap_system_view.ex

```diff
@@ -0,0 +1,157 @@
+    render_many(sap_systems, __MODULE__, "sap_system.json")
+  end
+
+  def render("sap_system.json", %{
+        sap_system:
+          %{
+            application_instances: application_instances,
+            database_instances: database_instances
+          } = sap_system
+      }) do
+    rendered_application_instances =
+      render_many(application_instances, __MODULE__, "application_instance.json", as: :instance)
+
+    rendered_database_instances =
+      render_many(database_instances, __MODULE__, "database_instance.json", as: :instance)
+
+    sap_system
+    |> Map.from_struct()
+    |> Map.delete(:__meta__)
+    |> Map.put(
+      :database_instances,
+      rendered_database_instances
+    )
+    |> Map.put(
+      :application_instances,
+      rendered_application_instances
+    )
+    |> add_system_replication_status_to_secondary_instance
+  end
+
+  def render("sap_system_registered.json", %{sap_system: sap_system}) do
+    sap_system
+    |> Map.from_struct()
+    |> Map.delete(:__meta__)
+    |> Map.delete(:database_instances)
+    |> Map.delete(:application_instances)
+    |> Map.delete(:tags)
+  end
+
+  def render("sap_system_health_changed.json", %{health: health}), do: health
+
+  def render("database_instance.json", %{instance: instance}) do
+    instance
+    |> Map.from_struct()
+    |> Map.delete(:__meta__)
+    |> Map.delete(:host)
+  end
+
+  def render("application_instance.json", %{instance: instance}) do
+    instance
+    |> Map.from_struct()
+    |> Map.delete(:__meta__)
+    |> Map.delete(:host)
+  end
+
+  def render("application_instance_health_changed.json", %{health: health}), do: health
+
+  def render("database_registered.json", %{database: database}) do
+    database
+    |> Map.from_struct()
+    |> Map.delete(:__meta__)
+    |> Map.delete(:tags)
+    |> Map.delete(:database_instances)
+  end
+
+  def render("database_health_changed.json", %{health: health}), do: health
+
+  def render("database_instance_health_changed.json", %{
+        instance: %{
+          sap_system_id: sap_system_id,
+          host_id: host_id,
+          instance_number: instance_number,
+          health: health
+        }
+      }) do
+    %{
+      sap_system_id: sap_system_id,
+      host_id: host_id,
+      instance_number: instance_number,
+      health: health
+    }
+  end
+
+  def render("database_instance_system_replication_changed.json", %{
+        instance: %{
+          sap_system_id: sap_system_id,
+          host_id: host_id,
+          instance_number: instance_number,
+          system_replication: system_replication,
+          system_replication_status: system_replication_status
+        }
+      }) do
+    %{
+      sap_system_id: sap_system_id,
+      host_id: host_id,
+      instance_number: instance_number,
+      system_replication: system_replication,
+      system_replication_status: system_replication_status
+    }
+  end
+
+  defp add_system_replication_status_to_secondary_instance(
```

Comment:
I'm not sure we should move this here. This looks an important domain logic component, not simply a view thing. Maybe we want to have this information somewhere else.

Reply:
I don't know, in the old code this was used only when the function in which was called was used by the controller, so I assumed it was only related to the view section, we can certainly move that if we think it's valuable

Reply:
Ok, I have checked the code and now understand what's going on.
The thing is that we do something strange in the read models projection (and in our domain logic in general). Having an `ACTIVE` or not sync status usually is attached to the secondary node, which is the node that is doing the sync (so this is the one that is active).
In the other hand, due how we handle the data (and how is sent by some HANA native outputs), we attach this value to the primary, and it is stored this way in the read model.
That's why we do this transition in the view to put the state in the secondary.

So long story short, how it is done, yes, this code belongs to the view, because we simply use it to display correctly things.
But I find that we could do this better, and store the `status` correctly in the projections, to avoid having this strange behaviour.
It is in any case a low hanging fruit. I will create tech-debt ticket myself.
So, you can let this code as you changed XD